code/datums/components/deadchat_control.dm

```diff
@@ -0,0 +1,134 @@
+#define DEMOCRACY_MODE "democracy"
+#define ANARCHY_MODE "anarchy"
+
+/datum/component/deadchat_control
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/list/inputs = list("up", "down", "left", "right", "north", "south", "west", "east")
+	var/list/ckey_to_cooldown = list()
+	var/deadchat_mode = DEMOCRACY_MODE
+	var/democracy_cooldown
+	var/anarchy_cooldown
+	var/force_move
+	var/require_orbiting
+
+/datum/component/deadchat_control/Initialize(_deadchat_mode, _democracy_cooldown = 12 SECONDS, _anarchy_cooldown = 20 SECONDS, _force_move = TRUE, _require_orbiting = FALSE)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(SSdcs, COMSIG_GLOB_DEAD_SAY, .proc/deadchat_react)
```

Comment:
We shouldn't be encouraging spam in the chat, make an actual basic ui while they orbit the thing or something. Yes that means you *have* to orbit rather than it being optional but so be it.

Alternatively, intercept all attempts by ghosts to say a direction and just cancel them saying it. I prefer the first option but this is fine too.

Reply:
Ill probably make it so it is either global or not global, and not global opens the UI.